framework_tvb/tvb/interfaces/web/controllers/simulator/simulator_controller.py

```diff
@@ -238,13 +237,14 @@ def set_surface(self, **data):
     @expose_fragment('simulator_fragment')
     def set_cortex(self, **data):
         session_stored_simulator, is_simulation_copy, is_simulation_load, _ = self.context.get_common_params()
+        rm_fragment = SimulatorRMFragment()
 
         if cherrypy.request.method == POST_REQUEST:
             self.context.add_last_loaded_form_url_to_session(SimulatorWizzardURLs.SET_STIMULUS_URL)
-            rm_fragment = SimulatorRMFragment()
             rm_fragment.fill_from_post(data)
             rm_fragment.fill_trait(session_stored_simulator.surface)
 
+        self.range_parameters.surface_parameters = rm_fragment.get_range_parameters()
```

Comment:
Isn't the coupling_strength added to the PSE params list even if no surface has been selected for the simulation?

Reply:
It won't be added, because this setting is in the set_cortex method and that method is not called when a surface is not provided.

Reply:
After talking with Robert, we concluded that we need to handle more carefully the PSE params which come from optional fragments, otherwise they will be left in the PSE params list even if no surface/noise is chosen. He will add a fix soon.

Reply:
It should be okay now.